test_all.py

```diff
@@ -31,5 +33,27 @@ def test_chain():
     #Test if equal
     assert system_to_FermionOperator(syst) == ham
     
+def test_spin_hubbard():
+    
+    a = 1
+    L = 2
+    t = 1
+    
+    sigma_0 = tinyarray.array([[1, 0], [0, 1]])
+
+    #Build an LxL square lattice in kwant
+    lat = kwant.lattice.square(a)
+    syst = kwant.Builder()   
+    syst[(lat(x, y) for x in range(L) for y in range(L))] = 0 * sigma_0
+    ##Hopping terms between neighbours with the same spin
+    syst[lat.neighbors()] = -t * sigma_0
+    syst = syst.finalized()
+
+    #Build a hamiltonian in openfermion
+    ham = openfermion.hamiltonians.fermi_hubbard(L, L, t, 0, spinless = False)
```

Comment:
Might be better to explicitly construct this by hand, so that if something changes inside the fermi_hubbard function this test doesn't fail. Or better yet, run through some explicit checks of what you would expect system_to_FermionOperator(syst) to look like.